scripts/index.js

```diff
@@ -1 +1,367 @@
+
+        // > strMultiply('abc', 5)
+        // 'abcabcabcabcabc'
+    // You may not use the native repeat method that strings provide. But you can use the range function defined below.
+
+    // You may use the following range function as is:
+
+        // function range(min, max) {
+        //   const arr = [];
+        //   for (var i = min; i < max; i++) {
+        //     arr.push(i);
+        //   }
+        //   return arr;
+        // }
+//________________________________________________________________________________________________________________________________
+
+
+function strMultiply(str, frequency) {
+    let arr = ``
+    while (frequency > 0) {
+        arr += str;
+        frequency--;
+    }
+    return arr
+};
+
+console.log(`#9`);
+console.log(strMultiply(`abc`,5));
+
+
+
+// Bonus: Array sorting
+// ---------------------
+// For these exercises, you'll want to refer to the MDN docs on using Array's sort method.
+//________________________________________________________________________________________________________________________________
+
+
+
+// 10. Sort an array
+// -----------------
+// Given an array of strings such as the array of names given in one of the "Good Job" problem, sort them by alphabetically order.
+//________________________________________________________________________________________________________________________________
+
+function sortArray1(arr) {
+    return arr.sort();
+}
+
+console.log(`#10`);
+console.log(sortArray1(people));
+
+
+
+// 11. Sort an array, 2
+// --------------------
+// Sort the same array, but not by alphabetically order, but by how long each name is, shortest name first.
+//________________________________________________________________________________________________________________________________
+
+function byLength(a, b) {
+    if (a.length < b.length){
+        return -1;
+    } 
+    if (a.length > b.length){
+        return 1;
+    }
+    return 0
+};
+
+
+console.log(`#11`);
+console.log(people.sort(byLength)) ;
+// console.log(people.map(sortArray2(people)));
+
+
+// 12. Sort an array, 3
+// --------------------
+// Given an array of array of products:
+
+const products = [
+  { name: 'Basketball', price: 12.00 },
+  { name: 'Tennis Racquet', price: 66.00 },
+  { name: 'Tennis Balls', price: 9.00 },
+  { name: 'Tennis Balls', price: 9.00 }
+];
+
+// Sort the array by price.
+
+//________________________________________________________________________________________________________________________________
+function byPrice(a, b) {
+    if (a.price < b.price){
+        return -1;
+    } 
+    if (a.price > b.price){
+        return 1;
+    }
+    return 0
+};
+
+
+console.log(`#12`);
+console.log(products.sort(byPrice)) ;
+
+
+
+// Reduce Example:
+// --------------------
+const mealPrice = [10, 15, 18, 10, 12];
+
+function tax(total, price) {
+  return total + (price * 1.1);
+};
+
+console.log(`#reduce example`)
+console.log(mealPrice.reduce(tax));
```

Comment:
I'm  happy that this is here 👍 
You're very thorough with your note taking and with your comments, and that will help you tremendously as you progress as a developer!